session/session.go

```diff
@@ -566,6 +585,7 @@ func (s *session) retry(ctx context.Context, maxCnt uint) error {
 			}
 			err = s.StmtCommit()
 			if err != nil {
+				s.txn.reset()
```

Comment:
I think it's better not to maintain s.txn.state outside the `txn.go` file.
session should only care the exposed function like `StmtCommit/StmtRollback/Commit/Rollback`

If `s.StmtCommit` return error, `StmtCommit` function should guarantee that the statement related state is clean up.